src/main/java/thePackmaster/actions/conjurerpack/PlayRandomCardAction.java

```diff
@@ -0,0 +1,51 @@
+package thePackmaster.actions.conjurerpack;
+
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.actions.common.EmptyDeckShuffleAction;
+import com.megacrit.cardcrawl.actions.common.PlayTopCardAction;
+import com.megacrit.cardcrawl.actions.unique.RandomCardFromDiscardPileToHandAction;
+import com.megacrit.cardcrawl.actions.utility.NewQueueCardAction;
+import com.megacrit.cardcrawl.actions.utility.UnlimboAction;
+import com.megacrit.cardcrawl.actions.utility.WaitAction;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.cards.CardGroup;
+import com.megacrit.cardcrawl.core.AbstractCreature;
+import com.megacrit.cardcrawl.core.Settings;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+
+public class PlayRandomCardAction extends AbstractGameAction
+{
+    private final CardGroup group;
+    public PlayRandomCardAction(AbstractCreature target, CardGroup group, int amount) {
+        this.duration = Settings.ACTION_DUR_FAST;
+        this.actionType = ActionType.WAIT;
+        this.source = AbstractDungeon.player;
+        this.target = target;
+        this.amount = amount;
+        this.group = group;
+    }
+
+    public void update() {
+        if (group.size() == 0) {
+            this.isDone = true;
+            return;
+        }
+
+        CardGroup tmp = new CardGroup(CardGroup.CardGroupType.UNSPECIFIED);
+        tmp.group.addAll(group.group);
+
+        for (int i = 0; i < Math.min(tmp.size(), amount); i++)
+        {
+            AbstractCard c = tmp.getRandomCard(AbstractDungeon.cardRng);
+            if (c != null)
+            {
+                group.group.remove(c);
+                AbstractDungeon.getCurrRoom().souls.remove(c);
+                this.addToBot(new NewQueueCardAction(c, target, false, true));
```

Comment:
I had ended up making some additional changes because the previous version had a bug where cards could be chosen twice by the loop. Accordingly, the tmp group was never actually needed, so I removed it.

Finally, I decided to only set the target x/y variables so that the player would see the cards move from the discard pile to slightly left of the center of the screen. Would it be better to just have them start off in the center as was suggested above?

Reply:
Setting only target_x/y is fine too. As you've noted, that just means there's the card is moved from the discard to the center of the screen, so it's just a matter of preference for how you want the visual effect to look.